gateway/gateway.go

```diff
@@ -510,7 +510,12 @@ func (gateway *HandleT) userWebRequestWorkerProcess(userWebRequestWorker *userWe
 				anonIDFromReq := strings.TrimSpace(vjson.Get("anonymousId").String())
 				userIDFromReq := strings.TrimSpace(vjson.Get("userId").String())
 				if builtUserID == "" {
-					builtUserID = anonIDFromReq + DELIMITER + userIDFromReq
+					if anonIDFromReq != "" {
+						builtUserID = anonIDFromReq + DELIMITER + userIDFromReq
+					} else {
+						// Proxy gets the userID from body if there is no anonymousId in body/header
+						builtUserID = userIDFromReq + DELIMITER + anonIDFromReq
```

Comment:
Why are we using `anonIDFromReq` as the second part? Isn't using `userIDFromReq` in place of `anonIDFromReq` the only change that we need?
```suggestion
						builtUserID = userIDFromReq + DELIMITER + userIDFromReq
```

Reply:
Yeah true... we are not concerned about what the second value is in most of the places

Reply:
does the migrator only parse the 1st part?

Reply:
Yes,we use this function to determine which partition the user belongs to `pg_catalog.PartitionIndex(split_part("jobs"."user_id", '<<>>', 1)`  in the SQL query

Reply:
Don't we need to introduce an SQL migration script as well so that UserIDs of existing events in jobsdb tables are renamed accordingly? Otherwise we might experience ordering inconsistencies, between old and new events. 
What about backed-up data? Won't we observe ordering inconsistencies if we replay half events with the old format and half events with the new format? Is it worth to address this as well?

Reply:
As an alternative, the migrator could use the first non-empty part of the composite key, instead of always using the 1st part, wdyt?

Reply:
I guess we wouldn't need to have migration scripts to address this, Also this would take a bit of time before we migrate enterprise customers into the migrator setup and we back up only 1-month of data so I guess data would get clean by then and for pro customers, we do support complete event replay and only replay errors. So I guess event ordering would be lost there by default

Reply:
So after deploying the new rudder-server version containing this change, we accept the fact that events with no anonymousId which already existed in jobsdb tables might be processed along with new events arriving to the server, in mixed order, due to this change, correct?